topo/node/cptx/cptx.go

```diff
@@ -26,18 +38,132 @@ func New(pb *topopb.Node) (node.Implementation, error) {
 
 type Node struct {
 	pb *topopb.Node
+	cliConn *scraplinetwork.Driver
 }
 
 func (n *Node) Proto() *topopb.Node {
 	return n.pb
 }
 
+// WaitCLIReady attempts to open the transport channel towards a Network OS and perform scrapligo OnOpen actions
+// for a given platform. Retries indefinitely till success.
+func (n *Node) WaitCLIReady() error {
+	transportReady := false
+	for !transportReady {
+		if err := n.cliConn.Open(); err != nil {
+			log.Debugf("%s - cli not ready - waiting.", n.pb.Name)
+			time.Sleep(time.Second * 2)
+			continue
+		}
+		transportReady = true
+		log.Debugf("%s - cli ready.", n.pb.Name)
+	}
+
+	return nil
+}
+
+// PatchCLIConnOpen sets the OpenCmd and ExecCmd of system transport to work with `kubectl exec` terminal.
+func (n *Node) PatchCLIConnOpen(ns string) error {
+	t, ok := n.cliConn.Transport.Impl.(scraplitransport.SystemTransport)
+	if !ok {
+		return ErrIncompatibleCliConn
+	}
+
+	t.SetExecCmd("kubectl")
+	t.SetOpenCmd([]string{"exec", "-it", "-n", ns, n.pb.Name, "--", "cli", "-c"})
+
+	return nil
+}
+
+// SpawnCLIConn spawns a CLI connection towards a Network OS using `kubectl exec` terminal and ensures CLI is ready
+// to accept inputs.
+func (n *Node) SpawnCLIConn(ns string) error {
+	d, err := scraplicore.NewCoreDriver(
+		n.pb.Name,
+		"juniper_junos",
```

Comment:
Is there a reason why juniper junos is hardcoded ?

Reply:
This is how scrapligo factory gets to know which vendor related driver to instantiate.

https://github.com/scrapli/scrapligo/blob/main/driver/core/factory.go

``` // SupportedPlatforms pseudo constant providing slice of all core platform types.
func SupportedPlatforms() []string {
    return []string{"cisco_iosxe",
        "cisco_iosxr",
        "cisco_nxos",
        "arista_eos",
        "juniper_junos",
        "nokia_sros",
        "nokia_sros_classic",
    }
}
```